src/main.py

```diff
@@ -172,22 +173,27 @@ async def fetch_ingest_lines_task(context: MetricsContext, project_id: str, serv
     topology_tasks = []
     topology_task_services = []
     skipped_topology_services = set()
-
-    for service in services:
-        if service.name in entities_extractors:
-            if entities_extractors[service.name].used_api in disabled_apis:
-                skipped_topology_services.add(service.name)
-                continue
-            topology_task = entities_extractors[service.name].extractor(context, project_id, service)
-            topology_tasks.append(topology_task)
-            topology_task_services.append(service)
-
-    if skipped_topology_services:
-        skipped_topology_services_string = ", ".join(skipped_topology_services)
-        context.log(project_id, f"Skipped fetching topology for disabled services: {skipped_topology_services_string}")
-
-    fetch_topology_results = await asyncio.gather(*topology_tasks, return_exceptions=True)
-
+    fetch_topology_results = ()
+
+    # Using metrics scope feature, fetching topology is not needed
+    if not ConfigurationParameters.scoping_project_support_enabled:
+        for service in services:
```

Comment:
This block looks like candidate for method extraction

Reply:
There are some minor impl alternatives here, but I extracted it.